Dockerfile

```diff
@@ -1,10 +1,12 @@
-FROM python:3.6
+FROM python:3.6-alpine
 
 RUN mkdir -p /usr/src/app/requirements
 WORKDIR /usr/src/app
 
 ADD requirements/requirements.txt requirements/
-RUN pip install -r requirements/requirements.txt
+RUN apk --no-cache add postgresql-dev gcc musl-dev && \
```

Comment:
Wouldn't the apk install be better as a layer before adding the requirements file? Lave just the pip install and the apk del for this layer

Reply:
:thinking: 
The rationale here was that changes to requirements.txt might lead to more deps being installed (and removed?) via apk. But I see your point.

I'm also having all "empty" layers (`ENV`, `CMD`, `EXPOSE`) brought to the top as well.

Reply:
Yeah, but if changes to the requirements require more system deps, the preceding layer might be changed as well, as the requirements file won't carry this dependencies. And it is more probable that "pure python" requirements are added, leading to more changes happening on the latter.